src/main/java/io/cdap/plugin/couchbase/CouchbaseSourceConfig.java

```diff
@@ -0,0 +1,359 @@
+  @Macro
+  private int timeout;
+
+  public CouchbaseSourceConfig(String referenceName, String nodes, String bucket, String selectFields,
+                               String conditions, String user, String password, String onError, String schema,
+                               int maxParallelism, String consistency, int timeout) {
+    this.referenceName = referenceName;
+    this.nodes = nodes;
+    this.bucket = bucket;
+    this.selectFields = selectFields;
+    this.conditions = conditions;
+    this.user = user;
+    this.password = password;
+    this.onError = onError;
+    this.schema = schema;
+    this.maxParallelism = maxParallelism;
+    this.consistency = consistency;
+    this.timeout = timeout;
+  }
+
+  public String getReferenceName() {
+    return referenceName;
+  }
+
+  public String getNodes() {
+    return nodes;
+  }
+
+  public String getBucket() {
+    return bucket;
+  }
+
+  public String getSelectFields() {
+    return selectFields;
+  }
+
+  @Nullable
+  public String getConditions() {
+    return conditions;
+  }
+
+  @Nullable
+  public String getUser() {
+    return user;
+  }
+
+  @Nullable
+  public String getPassword() {
+    return password;
+  }
+
+  public String getOnError() {
+    return onError;
+  }
+
+  public String getSchema() {
+    return schema;
+  }
+
+  public int getMaxParallelism() {
+    return maxParallelism;
+  }
+
+  public String getConsistency() {
+    return consistency;
+  }
+
+  public int getTimeout() {
+    return timeout;
+  }
+
+  /**
+   * Parses the json representation into a schema object.
+   *
+   * @return parsed schema object of json representation.
+   * @throws RuntimeException if there was an exception parsing the schema.
+   */
+  @Nullable
+  public Schema getParsedSchema() {
+    try {
+      return schema == null ? null : Schema.parseJson(schema);
+    } catch (IOException e) {
+      // this should not happen, since schema string comes from UI
+      throw new IllegalStateException(String.format("Could not parse schema string: '%s'", schema), e);
+    }
+  }
+
+  public List<String> getNodeList() {
+    return Arrays.asList(getNodes().split(","));
+  }
+
+  public List<String> getSelectFieldsList() {
+    return Arrays.asList(getSelectFields().split(","));
+  }
+
+  public Consistency getScanConsistency() {
+    return Objects.requireNonNull(Consistency.fromDisplayName(consistency));
+  }
+
+  public ErrorHandling getErrorHandling() {
+    return Objects.requireNonNull(ErrorHandling.fromDisplayName(onError));
+  }
+
+  /**
+   * Validates {@link CouchbaseSourceConfig} instance.
+   *
+   * @param collector failure collector.
+   */
+  public void validate(FailureCollector collector) {
+    if (Strings.isNullOrEmpty(referenceName)) {
+      collector.addFailure("Reference name must be specified", null)
+        .withConfigProperty(Constants.Reference.REFERENCE_NAME);
+    } else {
+      try {
+        IdUtils.validateId(referenceName);
```

Comment:
use the method that also takes a FailureCollector as an argument.

Reply:
Changed to use `IdUtils#validateReferenceName` method, which accepts a FailureCollector as an argument.